generic/proof-shell.el

```diff
@@ -738,6 +745,11 @@ unless the FLAGS for the command are non-nil (see `proof-action-list')."
     ;; proof-action-list is empty on error.
     (setq proof-action-list nil)
     (proof-release-lock)
+    (unless proof-shell-busy
```

Comment:
I am not sure this test is necessary. Why would the shell be busy if an error is detected? Probably not harmful.

Reply:
I left it in, but can remove if you'd like.